enclave/src/main/java/org/hyperledger/besu/enclave/VertxRequestTransmitter.java

```diff
@@ -75,24 +77,34 @@ public VertxRequestTransmitter(final HttpClient httpClient) {
       final boolean withAcceptJsonHeader) {
     try {
       final CompletableFuture<T> result = new CompletableFuture<>();
-      final HttpClientRequest request =
-          client
-              .request(method, endpoint)
-              .handler(response -> handleResponse(response, responseHandler, result))
-              .setTimeout(REQUEST_TIMEOUT_MS)
-              .exceptionHandler(result::completeExceptionally)
-              .setChunked(false);
+      RequestOptions options = new RequestOptions();
+      options.setTimeout(REQUEST_TIMEOUT_MS);
+      options.setMethod(method);
+      options.setURI(endpoint);
       if (withAcceptJsonHeader) {
         // this is needed when using Tessera GET /transaction/{hash} to choose the right RPC
-        request.putHeader(HttpHeaderNames.ACCEPT, APPLICATION_JSON);
+        options.putHeader(HttpHeaderNames.ACCEPT, APPLICATION_JSON);
       }
-      contentType.ifPresent(ct -> request.putHeader(HttpHeaders.CONTENT_TYPE, ct));
+      contentType.ifPresent(ct -> options.putHeader(HttpHeaders.CONTENT_TYPE, ct));
+
```

Comment:
just a consideration, about the new approach of Vert.x 4 that is harder to read and more verbose that the previous one :(

Reply:
Using the WebClient instead of HttpClient would be more readable, but I was wary of making a more dramatic change for the upgrade.